ocis-pkg/sync/nrwmutex.go

```diff
@@ -0,0 +1,49 @@
+package sync
+
+import (
+	"sync"
+)
+
+// NRWMutex works the same as RWMutex, the only difference is that it stores mutexes in a map and reuses them.
+// It's handy if you want to write-lock, write-unlock, read-lock and read-unlock for specific names only.
+type NRWMutex struct {
```

Comment:
Is NRWMutex a common name / convention? If not I would suggest switching to `NameRWMutex`